torch/csrc/jit/export.cpp

```diff
@@ -199,19 +201,20 @@ void EncoderBase::EncodeValueInfo(
     onnx::ValueInfoProto* v,
     const Value* n) {
   v->set_name(n->uniqueName());
-  onnx::TypeProto* t = v->mutable_type();
-  onnx::TypeProto_Tensor* tensor_type = t->mutable_tensor_type();
-
-  onnx::TensorShapeProto* shape = tensor_type->mutable_shape();
   if (CompleteTensorTypePtr node_type = n->type()->cast<CompleteTensorType>()) {
+    onnx::TypeProto* t = v->mutable_type();
+    onnx::TypeProto_Tensor* tensor_type = t->mutable_tensor_type();
+    onnx::TensorShapeProto* shape = tensor_type->mutable_shape();
     const std::vector<std::int64_t>& sizes = node_type->sizes();
     for (size_t i = 0; i < sizes.size(); i++) {
       shape->add_dim();
       shape->mutable_dim(i)->set_dim_value(sizes[i]);
     }
     tensor_type->set_elem_type(ATenTypeToOnnxType(node_type->scalarType()));
-  } else {
-    tensor_type->set_elem_type(onnx::TensorProto_DataType_UNDEFINED);
```

Comment:
could you still preserve this behavior? (set the elem_type to undefined if it's neither complete tensor nor bool.)

Reply:
Is this required for caffe2? 'undefined' type is messing up type inference in onnxruntime. This flag doesn't seem very useful as it's not providing any extra information than not setting the type at all.

Reply:
okay, at least, can we just error out if you think should not have undefined type here. IMO, blindly setting all the remaining type as bool is inappropriate behavior.

Reply:
it's only setting booltype when the node is BoolType(scalar). For the rest the type is left empty, as opposed to undefined.